extras/create_send_notifications.py

```diff
@@ -42,26 +42,49 @@ def create_output_directories():
         os.path.join(NOTIFICATIONS_BASE_DIR, NOTIFICATION_ARCHIVE_DIR)
     )
 
+def build_notifications_org_list(db):
+    """Return notifications organization list.
 
-def build_cyhy_org_list(db):
-    """Build list of CyHy organization IDs.
-
-    This is the list of CyHy organization IDs (and their descendants) that
-    receive CyHy reports.
+    This is the list of organization IDs that should
+    have a notification generated and sent.
+    """
+    notifications_to_generate = set()
+    cyhy_parent_ids = set()
+    ticket_owner_ids = db.NotificationDoc.collection.distinct("ticket_owner")
+    for request in db.RequestDoc.collection.find({"_id": {"$in": ticket_owner_ids}, "report_types": "CYHY"}, {"_id":1}):
+        # If the notification document's ticket owner has "CYHY" in their list of report_types,
+        # then a notification should be generated for that owner:
+        notifications_to_generate.add(request["_id"])
+        logging.debug("{} - Log request id".format(request["_id"]))
+        # Recursively check for any parents of the ticket owner that have "CYHY" in
+        # their list of report_types.  If found, add them to the list of owners that
+        # should get a notification.
+        cyhy_parent_ids.update(find_cyhy_parents(db, request["_id"]))
+    notifications_to_generate.update(cyhy_parent_ids)
+    notifications_to_delete = set(ticket_owner_ids) - notifications_to_generate
+    return list(notifications_to_generate), list(notifications_to_delete)
+          
+def find_cyhy_parents(db, org_id):
+    """Return parents/grandparents/etc. of an organization that have "CYHY" in their list of report_types.
     """
-    cyhy_org_ids = set()  # Use a set here to avoid duplicates
-    for cyhy_request in list(
-        db.RequestDoc.collection.find(
-            {"report_types": "CYHY"}, {"_id": 1, "children": 1}
-        ).sort([("_id", 1)])
-    ):
-        cyhy_org_ids.add(cyhy_request["_id"])
-        if cyhy_request.get("children"):
-            cyhy_org_ids.update(db.RequestDoc.get_all_descendants(cyhy_request["_id"]))
-    return list(cyhy_org_ids)
-
-
-def generate_notification_pdfs(db, org_ids, master_report_key):
+    cyhy_parents = set()
+    for request in db.RequestDoc.collection.find({"children": org_id}, {"_id": 1, "report_types": 1}):
+        if "CYHY" in request["report_types"]:
+            # There is an undocumented constraint where organizations are set up
+            # with only one level of children (i.e. no grandchildren orgs). Given 
+            # the fact that it is undocumented, following the hierarchy to the top
+            # is the best solution. So we do not exepct to go up more than one level.
+            # However, the implementation above is just a safety measure.
```

Comment:
Here's my suggested edit for this comment:
```suggestion
            # There is an undocumented convention at CISA to set up CyHy
            # organizations with only one level of children (i.e. no
            # grandchildren orgs). Since it is only a convention and not an
            # enforced rule, we decided that following the hierarchy to the top
            # is the safest solution.
```

Reply:
I think what @dav3r wrote is better than what I wrote.

Reply:
@dav3r, @jsf9k please see https://github.com/cisagov/cyhy-reports/pull/96/commits/5335968f236f6ec39b78ebc7484561b4e85e782b